### PR TITLE
Disable missing rule check for the version lock

### PR DESCRIPTION
**Pull request by rw-access**
## Issues
There was a failed 7.13 backport for #1383 because the lock file had a superset of rules.

## Summary
Disabled a test so the lock file can reference rules that don't exist.
Since we're revisiting the lock file soon, this won't be the long term solution.

### CODE REVIEW

**Comment by brokensound77** [comment]
Do you know why this failure is just surfacing now? These rules have been deprecated for a while and we have done several locks and backports since then

**Comment by rw-access** [comment]
the version lock contains 7.14 only rules, which 7.13 thinks is an error

**Comment by brokensound77** [comment]
> the version lock contains 7.14 only rules, which 7.13 thinks is an error

ahh ok, thanks - these are the _new_ rules

File: tests/test_all_rules.py [hunk 1]
```diff
@@ -402,12 +402,14 @@ def test_deprecated_rules(self):
                 err_msg = f'{self.rule_str(rule)} deprecation_date and updated_date should match'
                 self.assertEqual(meta.deprecation_date, meta.updated_date, err_msg)
 
-        missing_rules = sorted(set(versions).difference(set(self.rule_lookup)))
-        missing_rule_strings = '\n '.join(f'{r} - {versions[r]["rule_name"]}' for r in missing_rules)
-        err_msg = f'Deprecated rules should not be removed, but moved to the rules/_deprecated folder instead. ' \
-                  f'The following rules have been version locked and are missing. ' \
-                  f'Re-add to the deprecated folder and update maturity to "deprecated": \n {missing_rule_strings}'
-        self.assertEqual([], missing_rules, err_msg)
+        # skip this so the lock file can be shared across branches
```
Comment (brokensound77) [inline tests/test_all_rules.py:405]:
```suggestion
        # TODO: recheck this in some way, once version lock process is revised
        # skip this so the lock file can be shared across branches
```

**Review by brokensound77 — APPROVED** [review]
This works as a short term solution, but we should make sure we re-implement it in a way that makes sense once version lock changes.

Thanks 👍

**Pull request merged**